src/store/preferences.ts

```diff
@@ -206,6 +214,16 @@ export const actions = {
 
     return severities;
   },
+  async setShowMuted({ dispatch, rootState }: PrefActionContext, isMuted: boolean) {
+    await dispatch(
+      'preferences/commitPreferences',
+      {
+        ...rootState.credentials.credentials as ServerState,
+        payload: { diagnostics: { showMuted: isMuted } } as Partial<Settings>,
```

Comment:
```suggestion
        payload: { diagnostics: { showMuted: isMuted } } as RecursivePartial<Settings>,
```
(Actually, with the fix for the definition of `CommitArgs` the cast should be unnecessary.)

Reply:
This is true, removed

Reply:
NVM, maybe not. Looks like my editor wasn't picking up on the deprecation warning, but the linter definitely is..